### PR TITLE
Users/mizho/dropplugins

### PR DESCRIPTION
**Pull request by mizho**

### CODE REVIEW

File: Tasks/Checkout/task.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (TingluoHuang) [inline Tasks/Checkout/task.json]: remove this file, this is an example for you, not for final check-in  :D

File: Tasks/DownloadBuildDropV0/task.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  "helpMarkDown": "",
+  "category": "Utility",
+  "author": "Microsoft Corporation",
+  "version": {
+    "Major": 0,
+    "Minor": 137,
+    "Patch": 0
+  },
+  "groups": [],
+  "demands": [],
```
Comment (TingluoHuang) [inline Tasks/DownloadBuildDropV0/task.json]: you need demand min agent version for your new tasks.

File: Tasks/DownloadBuildDropV0/task.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      "type": "string",
+      "label": "The name of artifact to download.",
+      "required": true,
+      "helpMarkDown": "The name of artifact to download. The artifact must be a build drop."
+    },
+    {
+      "name": "targetpath",
+      "type": "filePath",
+      "label": "Path to download to",
+      "defaultValue": "$(Build.DropDownloadingDirectory)",
```
Comment (TingluoHuang) [inline Tasks/DownloadBuildDropV0/task.json]: why default to this variable? is this a predefined variable?

File: Tasks/DownloadBuildDropV0/task.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{
+  "id": "61F2A582-95AE-4948-B34D-A1B3C4F6A737",
+  "name": "DownloadBuildDrop",
+  "friendlyName": "Download Build Drop",
+  "description": "Download Build Drop",
+  "helpMarkDown": "",
```
Comment (TingluoHuang) [inline Tasks/DownloadBuildDropV0/task.json]: you might want to fill out helpMarkDown

File: Tasks/DownloadBuildDropV0/task.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (TingluoHuang) [inline Tasks/DownloadBuildDropV0/task.json]: mark this task as preview

File: Tasks/PublishBuildDropV0/task.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (TingluoHuang) [inline Tasks/PublishBuildDropV0/task.json]: mark as preview

File: Tasks/PublishBuildDropV0/task.json [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  "id": "ECDC45F6-832D-4AD9-B52B-EE49E94659BE",
+  "name": "PublishBuildDrop",
+  "friendlyName": "Publish Build Drop",
+  "description": "Publish Build Drop",
+  "helpMarkDown": "",
```
Comment (TingluoHuang) [inline Tasks/PublishBuildDropV0/task.json]: you might want to fill out helpMarkDown

File: Tasks/PublishBuildDropV0/task.json [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  "category": "Utility",
+  "author": "Microsoft Corporation",
+  "version": {
+    "Major": 0,
+    "Minor": 137,
+    "Patch": 0
+  },
+  "groups": [],
+  "demands": [],
```
Comment (TingluoHuang) [inline Tasks/PublishBuildDropV0/task.json]: min agent version demands.

File: Tasks/PublishBuildDropV0/task.json [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  "inputs": [
+    {
+      "name": "artifactname",
+      "type": "string",
+      "label": "The name of this artifact",
+      "defaultValue": "",
```
Comment (TingluoHuang) [inline Tasks/PublishBuildDropV0/task.json]: i thought we have a default artifact name "drop" today, do we?

File: Tasks/Checkout/task.json (hunk 1 above)
Comment (mizho) [reply Tasks/Checkout/task.json]: removed

File: Tasks/DownloadBuildDropV0/task.json (hunk 2 above)
Comment (mizho) [reply Tasks/DownloadBuildDropV0/task.json]: Is it something like this?

    "minimumAgentVersion": "1.91.0",

But what version should I use?

File: Tasks/DownloadBuildDropV0/task.json (hunk 3 above)
Comment (mizho) [reply Tasks/DownloadBuildDropV0/task.json]: removed

File: Tasks/DownloadBuildDropV0/task.json (hunk 4 above)
Comment (mizho) [reply Tasks/DownloadBuildDropV0/task.json]: done

File: Tasks/DownloadBuildDropV0/task.json (hunk 5 above)
Comment (mizho) [reply Tasks/DownloadBuildDropV0/task.json]: how?

File: Tasks/PublishBuildDropV0/task.json [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+      "required": true,
+      "helpMarkDown": "The name of this artifact."
+    },
+    {
+      "name": "targetpath",
+      "type": "filePath",
+      "label": "Path to publish",
+      "defaultValue": "$(Build.DropStagingDirectory)",
```
[the remark this answers is not in the document]
Comment (mizho) [reply Tasks/PublishBuildDropV0/task.json]: removed

File: Tasks/PublishBuildDropV0/task.json (hunk 9 above)
Comment (mizho) [reply Tasks/PublishBuildDropV0/task.json]: added

File: Tasks/PublishBuildDropV0/task.json (hunk 7 above)
Comment (mizho) [reply Tasks/PublishBuildDropV0/task.json]: added

File: Tasks/DownloadBuildDropV0/task.json (hunk 5 above)
Comment (TingluoHuang) [reply Tasks/DownloadBuildDropV0/task.json]: https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/AzureFileCopyV2/task.json#L19

File: Tasks/DownloadBuildDropV0/task.json (hunk 2 above)
Comment (TingluoHuang) [reply Tasks/DownloadBuildDropV0/task.json]: you might want to target next agent release. do 2.138.0

File: Tasks/DownloadBuildDropV0/task.json (hunk 5 above)
Comment (mizho) [reply Tasks/DownloadBuildDropV0/task.json]: done

File: Tasks/PublishBuildDropV0/task.json (hunk 8 above)
Comment (mizho) [reply Tasks/PublishBuildDropV0/task.json]: done

File: Tasks/PublishBuildDropV0/task.json (hunk 6 above)
Comment (mizho) [reply Tasks/PublishBuildDropV0/task.json]: done

File: Tasks/DownloadBuildDropV0/task.json (hunk 2 above)
Comment (mizho) [reply Tasks/DownloadBuildDropV0/task.json]: added

**Comment by TingluoHuang** [comment]
I am good with the task.json changes, double check with your PM make sure the input are right.

**Pull request merged**